spmd/compiler/api.py

```diff
@@ -433,11 +433,14 @@ def gather_inputs_for_compilation(
                 )
                 return compile_inps
 
-            self._compiled_m = aot_module(
+            self._compiled_m = aot_module_simplified(
                 self._local_module,
                 partial(self._compile, TrainingPhase.FORWARD),
                 partial(self._compile, TrainingPhase.BACKWARD),
-                pre_compile_fn=gather_inputs_for_compilation,
+                # pre_compile_fn=gather_inputs_for_compilation,
```

Comment:
Can we remove the comment here and add a comment below for why we need this? Also what are the downsides with this approach (if any)?